src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -384,4 +529,336 @@ private static String trimExcessWhitespaces(String text) {
+                    if (child.getType() == JavadocTokenTypes.TEXT) {
+                        result.append(child.getText());
+                    }
+                }
+            }
+        }
+        return result.toString().trim();
+    }
+
+    /**
+     * Get Description node from ast.
+     *
+     * @param ast Javadoc root node.
+     * @return Description DetailNode.
+     */
+    private static DetailNode getDescriptionNode(DetailNode ast) {
+        DetailNode descriptionNode = null;
+        for (DetailNode node : ast.getChildren()) {
+            if (node.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG) {
+                for (DetailNode dataType : node.getChildren()) {
+                    if (dataType.getType() == JavadocTokenTypes.DESCRIPTION) {
+                        descriptionNode = dataType;
+                    }
+                }
+            }
+        }
+        return descriptionNode;
+    }
+
+    /**
+     * Finds and returns first sentence.
+     *
+     * @param ast Javadoc root node.
+     * @return first sentence.
+     */
+    private static String getInlineFirstSentence(DetailNode ast) {
+        final StringBuilder result = new StringBuilder(256);
+        for (DetailNode node : ast.getChildren()) {
+            if (node.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG) {
+                result.append(returnInlineFirstSentence(node));
+            }
+
+            if (node.getType() == JavadocTokenTypes.HTML_ELEMENT) {
+                result.append(returnInlineFirstSentence(
+                        Objects.requireNonNull(getInlineTagNodeInsideHtml(ast))));
+            }
+        }
+        return result.toString();
+    }
+
+    /**
+     * Sets result string in method getInlineFirstSentence.
+     *
+     * @param node JavaDoc Inline node.
+     * @return first sentence.
+     */
+    public static String returnInlineFirstSentence(DetailNode node) {
+        final StringBuilder result = new StringBuilder(256);
+        final String periodSuffix = PERIOD;
+        for (DetailNode child : node.getChildren()) {
+            if (ALLOWED_TYPES.contains(child.getType())) {
+                final String text;
+                if (child.getChildren().length == 0) {
+                    text = child.getText();
+                }
+                else {
+                    text = getTextFromDescription(child);
+                }
+
+                if (text.contains(periodSuffix)) {
+                    result.append(text, 0, text.indexOf(periodSuffix) + 1);
+                    break;
+                }
+
+                result.append(text);
+            }
+        }
+
+        return result.toString();
+    }
+
+    /**
+     * Returns the text from an inline tag Description.
+     *
+     * @param ast Inline tag Node.
+     * @return Text of inline tag description.
+     */
+    private static String getTextFromDescription(DetailNode ast) {
+        final String period = PERIOD + ' ';
```

Comment:
there could be no space after period.
Please create test on this.

Reply:
Sir, there can be cases where there is space after period. Like - 
```
/**
 * {@summary Simple JavaDoc. }
 */
 private void foo1() {} // ok
```
But, here the `PERIOD` doesn't matter as the requirement of this function [`getTextFromDescription(DetailNode ast)`] is just to return the entire description. The actual requirement of `PERIOD` is to get the first sentence, which is done by `getFirstSentenceFromInlineTagNode()` and period is properly defined there as just `PERIOD`. I had removed the requirement of this period from `getTextFromDescription()` and had added a test case to verify all is right.